protocol/common/message_linkpreview.go

```diff
@@ -161,17 +171,23 @@ func (preview *StatusLinkPreview) validateForProto() error {
 	}
 
 	// At least and only one of Contact/Community/Channel should be present in the preview
-	if preview.Contact != nil && preview.Community != nil {
-		return fmt.Errorf("both contact and community are set at the same time")
+	var linkTypes []string
+	if preview.Contact != nil {
+		linkTypes = append(linkTypes, "Contact")
 	}
-	if preview.Community != nil && preview.Channel != nil {
-		return fmt.Errorf("both community and channel are set at the same time")
+	if preview.Community != nil {
+		linkTypes = append(linkTypes, "Community")
 	}
-	if preview.Channel != nil && preview.Contact != nil {
-		return fmt.Errorf("both contact and channel are set at the same time")
+	if preview.Channel != nil {
+		linkTypes = append(linkTypes, "Channel")
 	}
-	if preview.Contact == nil && preview.Community == nil && preview.Channel == nil {
-		return fmt.Errorf("none of contact/community/channel are set")
+	if preview.Transaction != nil {
+		linkTypes = append(linkTypes, "Transaction")
+	}
+	if len(linkTypes) > 1 {
+		return fmt.Errorf("multiple components set at the same time: %v", linkTypes)
+	} else if len(linkTypes) == 0 {
```

Comment:
No need for the else
```suggestion
	}
   if len(linkTypes) == 0 {
```